revocation/errors.go

```diff
@@ -0,0 +1,44 @@
+package revocation
+
+import "fmt"
+
+// RevokedInOCSPError is returned when the certificate's status for OCSP is ocsp.Revoked
+type RevokedInOCSPError struct{}
+
+func (e RevokedInOCSPError) Error() string {
+	return "certificate is revoked via OCSP"
+}
+
+// UnknownInOCSPError is returned when the certificate's status for OCSP is ocsp.Unknown
+type UnknownInOCSPError struct{}
+
+func (e UnknownInOCSPError) Error() string {
+	return "certificate has unknown status via OCSP"
+}
+
+// CheckOCSPError is returned when there is an error during the OCSP revocation check, not necessarily a revocation
+type CheckOCSPError struct {
+	Err error
+}
+
+func (e CheckOCSPError) Error() string {
+	if e.Err != nil {
+		return fmt.Sprintf("error checking revocation via OCSP: %s", e.Err.Error())
+	} else {
+		return "error checking revocation via OCSP"
+	}
+}
+
+// NoOCSPServerError is returned when the OCSPServer is not specified or is not an HTTP URL.
+type NoOCSPServerError struct{}
+
+func (e NoOCSPServerError) Error() string {
+	return "no valid OCSPServer found"
+}
```

Comment:
This would indicate that we aren't able to check OCSP revocation (since the OCSPServer for the cert isn't set or is LDAP instead of HTTP). However, it would be an issue with the cert itself, not with the request (making it different from  a few of the other errors)

Reply:
@priteshbandi does this answer your question? I can resolve this if so

Reply:
> the OCSPServer for the cert isn't set or is LDAP instead of HTTP)
Not having ocsp url set is different from having ldap url. If the url is ldap we should through an error or at minimum a warning.

Reply:
Makes sense. I'll create an InvalidOCSPServerError for such cases, but still preserve the NoOCSPServerError to indicate that OCSP cannot be checked. This will make it easier to differentiate between a good OCSP response of nil and a response that we'll need to check with CRL later

Reply:
I actually realized that the logic has since been changed to return a CheckOCSPError if there is a server that is not accessible via HTTP. So, there is no need for a new error type, the NoOCSPServerError is only returned if there is truly no OCSPServer for the cert